src/main/java/com/alistats/discorki/listener/SlashCommandListener.java

```diff
@@ -0,0 +1,200 @@
+import org.springframework.stereotype.Service;
+
+import com.alistats.discorki.controller.LeagueApiController;
+import com.alistats.discorki.dto.riot.league.LeagueEntryDto;
+import com.alistats.discorki.dto.riot.summoner.SummonerDto;
+import com.alistats.discorki.model.DiscordGuild;
+import com.alistats.discorki.model.DiscordUser;
+import com.alistats.discorki.model.Rank;
+import com.alistats.discorki.model.Summoner;
+import com.alistats.discorki.repository.DiscordGuildRepo;
+import com.alistats.discorki.repository.RankRepo;
+import com.alistats.discorki.repository.SummonerRepo;
+import com.alistats.discorki.repository.UserRepo;
+import com.alistats.discorki.view.DiscordLeaderboardView;
+
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+@Service
+public class SlashCommandListener extends ListenerAdapter {
+    @Autowired
+    private UserRepo userRepo;
+    @Autowired
+    private DiscordGuildRepo discordGuildRepo;
+    @Autowired
+    private RankRepo rankRepo;
+    @Autowired
+    private SummonerRepo summonerRepo;
+    @Autowired
+    private LeagueApiController leagueApiController;
+    @Autowired
+    private DiscordLeaderboardView discordLeaderboardView;
+
+    @Override
+    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
+        try {
+            switch (event.getName()) {
+                case "add":
+                    add(event);
+                    break;
+                case "remove":
+                    remove(event);
+                    break;
+                case "enable":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "disable":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "list_users":
+                    listUsers(event);
+                    break;
+                case "list_notifications":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "leaderboard":
+                    leaderboard(event);
+                    break;
+            }
+        } catch (Exception e) {
+            event.reply("An error occured.").queue();
+            e.printStackTrace();
+        }
+
+    }
+
+    private void add(SlashCommandInteractionEvent event) throws Exception {
+        // Get guild
+        DiscordGuild discordGuild = getOrCreateGuild(event.getGuild());
+        DiscordUser discordUser = new DiscordUser();
+
+        User user = event.getOption("user").getAsUser();
+        String summonerName = event.getOption("summoner_name").getAsString();
+
+        // Check if user is already in that guild
+        if (discordGuild.getUsers() != null
```

Comment:
Do these null checks have any added value? The response of getUsers() will be an empty array if the guild was just created and searching it will suffice.